src/manifests_base.py

```diff
@@ -251,3 +248,94 @@ def from_space_separated(cls, tolerations: str) -> List["CephToleration"]:
             return [cls._from_string(toleration) for toleration in tolerations.split()]
         except ValueError as e:
             raise ValueError(f"Invalid tolerations: {e}") from e
+
+
+class ProvisionerAdjustments(Patch):
+    """Update provisioner manifest objects."""
+
+    PROVISIONER_NAME: str
+    PLUGIN_NAME: str
+
+    def tolerations(self) -> Tuple[List[CephToleration], bool]:
+        return [], False
+
+    def adjust_container_specs(self, obj: AnyResource) -> None:
+        csidriver = cast(SafeManifest, self.manifests).csidriver
+        original_dn, updated_dn = csidriver.default_name, csidriver.formatted
+        kubelet_dir = self.manifests.config.get("kubelet_dir", "/var/lib/kubelet")
+
+        for c in obj.spec.template.spec.containers:
+            for idx in range(len(c.args)):
+                if original_dn in c.args[idx] and updated_dn not in c.args[idx]:
```

Comment:
This is by far my least favorite line.  

in the upstream manifests, there are container arguments that contain `*.csi.ceph.com`.  We need to change those argument strings to whatever the formatted name for the driver is. But we can only replace them once. Otherwise we might end up with:

```alt.alt.alt.alt.alt.alt.alt.alt.alt.rdb.csi.ceph.com```

if this method is called multiple times on the same resource.